booking-module/database/makeBookingCsv.js

```diff
@@ -0,0 +1,67 @@
+const fs = require('fs');
+const faker = require('faker');
+const moment = require('moment');
+const { Transform } = require('stream');
+
+// const { createInterface } = require('readline');
+
+const listings = fs.createReadStream('./listings.csv');
+
+const bookings = fs.createWriteStream('./bookings.csv');
+
+// const lineReader = createInterface({ input: listings });
+
+const createBooking = (idx, listingId, name) => {
+  const checkIn = moment(faker.date.between('2018-06-01', '2019-06-01')).format('YYYY-MM-DD');
+  const stayLength = Math.floor(Math.random() * 6) + 1;
+  const checkOut = moment(checkIn).add(stayLength, 'days').format('YYYY-MM-DD');
+  const guests = Math.floor(Math.random() * 6) + 1;
+  const user = faker.name.findName();
+  return `"${idx}","${listingId}","${name}","${checkIn}","${checkOut}","${guests}","${user}"\n`;
+};
+
+class BookingsGenerator extends Transform {
+  constructor(options) {
+    super(options);
+    this.cache = '';
+    this.idx = 1;
+  }
+
+  _transform(chunk, enc, cb) {
```

Comment:
I don't see `enc` being used anywhere in this method. Is that a necessary parameter?

Reply:
I don't use it. But you're never supposed to call _transform directly; it's a helper method that gets called by other, unseen, inherited methods in the class, and they pass it those three parameters. In my case, I know I'm only ever going to call this on a string so no need to deal with it. Mainly it's there so that cb is in the right place in the parameters list.

Though tbh I'm less than positive that I need to call the cb...